### DISCUSSION TITLE
Locale-aware number parsing

### DISCUSSION THREAD

**Discussion by RagibHasin** [unicode-org/icu4x#5501] [Q&A]
Is there a way to parse numbers from localized strings? Like parse 100 from `"১০০"` in locale `bn`?

**Comment by Manishearth** [comment]
Not at the moment, no. @sffc is this an API available in ICU4C?

I know in general we have avoided parsing APIs.

**Comment by sffc** [comment]
ICU4X and ECMA-402 do not currently support parsing of localized strings. Please read my blog post on the subject:

https://blog.sffc.xyz/post/190943794505/why-you-should-not-parse-localized-strings

We should, but I think we don't currently, export the Digit property so that you can write your own parser for your own use case.

As I've stated previously, I don't consider adding a "good" number or date parser to be out of scope. ICU4C's approach, essentially "backwards formatting", is very brittle, error-prone, and an excellent example of how _not_ to write a parser. The "best" parser would be to use an LLM, and if we were to implement one in ICU4X, I think it should start from the perspective of having behavior on par with what you would get from an LLM.

**Reply by RagibHasin** [reply]
Yes, having the Digit property would suffice, and maybe the best, so that downstream libraries can write their own parsers. How can I help in adding the Digit property?

**Reply by sffc** [reply]
Adding the Digit property would be very similar to when we added Hangul_Syllable_Type: https://github.com/unicode-org/icu4x/pull/4885

Take a look at the individual commits. You should be able to do something very similar to add the Digit property.
